src/AbstractResource.php

```diff
@@ -43,23 +41,41 @@ public function getAttributes(array $fields = null)
     }
 
     /**
+     * Get the links.
+     *
+     * @return array
+     */
+    public function getLinks()
+    {
+        return $this->links;
+    }
+    
+    /**
+     * Get the meta data.
+     *
+     * @return array
+     */
+    public function getMeta()
+    {
+        return $this->meta;
+    }
+
+    /**
      * {@inheritdoc}
      *
-     * @throws \LogicException
+     * @throws LogicException
      */
     public function getRelationship($name)
     {
         $method = $this->getRelationshipMethodName($name);
 
-        if (method_exists($this, $method)) {
-            $relationship = $this->$method();
+        $relationship = $this->$method();
 
-            if ($relationship !== null && ! ($relationship instanceof Relationship)) {
-                throw new LogicException('Relationship method must return null or an instance of Tobscure\JsonApi\Relationship');
-            }
-
-            return $relationship;
+        if ($relationship !== null && ! ($relationship instanceof Relationship)) {
```

Comment:
Just checked, it doesn't. Thus, the first part is needed to allow relationship methods to return null without this Exception being thrown.